applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -424,7 +424,7 @@ static void scan_connecting_error(struct bt_scan_device_info *device_info)
 {
 	LOG_WRN("Connecting failed");
 	scan_counter = SCAN_TRIG_TIMEOUT_MS;
-	k_delayed_work_submit(&scan_start_trigger,
```

Comment:
Ok. There is a potential race with scan_start and its cancel but I think it is impossible to occur as this callback will trigger when scanning was in progress and connection is attempted.